test/extended/networking/ovs.go

```diff
@@ -0,0 +1,314 @@
+					} else {
+						Fail("found unexpected OVS flow: " + flow)
+					}
+				} else {
+					newDeployNodeFlows = append(newDeployNodeFlows, flow)
+				}
+			}
+			Expect(arpOut).To(BeTrue(), "Should have an outgoing ARP rule")
+			Expect(arpIn).To(BeTrue(), "Should have an incoming ARP rule")
+			Expect(ipOut).To(BeTrue(), "Should have an outgoing IP rule")
+			if pluginIsolatesNamespaces() {
+				Expect(ipInGlobal && ipInIsolated).To(BeTrue(), "Should have global and isolated incoming IP rules")
+			} else {
+				Expect(ipInGeneric).To(BeTrue(), "Should have a generic incoming IP rule")
+			}
+			Expect(reflect.DeepEqual(origFlows[deployNodeName], newDeployNodeFlows)).To(BeTrue(), "Deploy node flows should be unchanged except for the new pod")
+
+			err := f1.Client.Pods(f1.Namespace.Name).Delete(podName, nil)
+			Expect(err).NotTo(HaveOccurred())
+
+			postDeleteFlows := getFlowsForNode(oc, deployNodeName)
+			Expect(reflect.DeepEqual(origFlows[deployNodeName], postDeleteFlows)).To(BeTrue(), "Flows after deleting pod should be same as before creating it")
+		})
+
+		It("should add and remove flows when nodes are added and removed", func() {
+			var err error
+			nodes := e2e.GetReadySchedulableNodesOrDie(f1.Client)
+			origFlows := getFlowsForAllNodes(oc, nodes.Items)
+
+			// The SDN/OVS code doesn't care that the node doesn't actually exist,
+			// but we try to pick an IP on our local subnet to avoid sending
+			// traffic into the real world.
+			highNodeIP := ""
+			for _, node := range nodes.Items {
+				if node.Status.Addresses[0].Address > highNodeIP {
+					highNodeIP = node.Status.Addresses[0].Address
+				}
+			}
+			Expect(highNodeIP).NotTo(Equal(""))
+			octets := strings.Split(highNodeIP, ".")
+			Expect(len(octets)).To(Equal(4))
+			lastOctet, err := strconv.Atoi(octets[3])
+			Expect(err).NotTo(HaveOccurred())
+			Expect(lastOctet).NotTo(Equal(255))
+			octets[3] = strconv.Itoa(lastOctet + 1)
+			newNodeIP := strings.Join(octets, ".")
+
+			nodeName := "ovs-test-node"
+			node := &kapi.Node{
+				TypeMeta: kapiunversioned.TypeMeta{
+					Kind: "Node",
+				},
+				ObjectMeta: kapi.ObjectMeta{
+					Name: nodeName,
+				},
+				Spec: kapi.NodeSpec{
+					Unschedulable: true,
+				},
+				Status: kapi.NodeStatus{
+					Addresses: []kapi.NodeAddress{
+						{
+							Type:    kapi.NodeInternalIP,
+							Address: newNodeIP,
+						},
+					},
+				},
+			}
+			node, err = f1.Client.Nodes().Create(node)
+			Expect(err).NotTo(HaveOccurred())
+
+			osClient, err := testutil.GetClusterAdminClient(testexutil.KubeConfigPath())
+			Expect(err).NotTo(HaveOccurred())
+
+			var hs *sdnapi.HostSubnet
+			e2e.Logf("Waiting up to %v for HostSubnet to be created", hostSubnetTimeout)
+			for start := time.Now(); time.Since(start) < hostSubnetTimeout; time.Sleep(time.Second) {
+				hs, err = osClient.HostSubnets().Get(node.Name)
+				if err == nil {
+					break
+				}
+			}
+			Expect(hs).NotTo(Equal(nil))
+			Expect(hs.Name).To(Equal(nodeName))
```

Comment:
We expect that but we don't need to Expect() it. The point of this check is that for some reason, on error osClient.HostSubnets().Get() returns `&HostSubnet{}` rather than `nil`, so the !nil check isn't sufficient to make sure it succeeded. Maybe I should check "err == nil" instead or something